components/brave_wallet/browser/internal/hd_key.h

```diff
@@ -56,10 +58,9 @@ class HDKey : public HDKeyBase {
       const std::vector<uint8_t>& ephemeral_public_key,
       const std::vector<uint8_t>& ciphertext) const;
 
-  // index should be 0 to 2^32
-  // 0 to 2^31-1 is normal derivation and 2^31 to 2^32-1 is harden derivation
-  // If anything failed, nullptr will be returned
-  std::unique_ptr<HDKeyBase> DeriveChild(uint32_t index) override;
+  std::unique_ptr<HDKeyBase> DeriveNormalChild(uint32_t index) override;
+  std::unique_ptr<HDKeyBase> DeriveHardenedChild(uint32_t index) override;
```

Comment:
DeriveNormalChild and DeriveHardenedChild are public and available through base class. DeriveChild is now private implementation for deriving secp256k1 keys.
DeriveNormalChild is not implemented for Ed25519 keys.